### PR TITLE
feat(number): add binary and octal random number generation

### PR DESCRIPTION
**Pull request by pladreyt**
This PR implements binary and octal random number generation.

**Issue:** https://github.com/faker-js/faker/issues/184

PR that implements this feature in string module: https://github.com/faker-js/faker/pull/1710

### CODE REVIEW

**Review by ST-DDT — COMMENTED** [review]
Could you please move the methods above the hex method so that the methods are a bit more in order.
Binary(2), Octal(8), Hex(16)

Should we also create these methods in the stri g module to generate them in a specific length?

File: test/number.spec.ts [hunk 1]
```diff
@@ -348,5 +350,71 @@ describe('number', () => {
         );
       });
     });
+
+    describe('binary', () => {
+      it('generates single binary character when no additional argument was provided', () => {
+        const binary = faker.number.binary();
+        expect(binary).toBeTypeOf('string');
+        expect(binary).toHaveLength(1);
+        expect(/^[01]+$/.test(binary)).toBe(true);
```
Comment (ST-DDT) [inline test/number.spec.ts]: Please use expect(binary).toMatch here and elsewhere.

File: test/number.spec.ts [hunk 2]
```diff
@@ -348,5 +350,71 @@ describe('number', () => {
...
+
+    describe('octal', () => {
+      it('generates single octal character when no additional argument was provided', () => {
+        const octal = faker.number.octal();
+        expect(octal).toBeTypeOf('string');
+        expect(octal).toHaveLength(1);
+        expect(/^[0-7]+$/.test(octal)).toBe(true);
+      });
+
+      it('generates a random octal string', () => {
```
Comment (ST-DDT) [inline test/number.spec.ts]: The title is duplicate and doesnt reflect the purpose of the test.
Can you please adjust it to something like.
Generates a random octal value with a custom max

File: test/number.spec.ts [hunk 3]
```diff
@@ -348,5 +350,71 @@ describe('number', () => {
...
+
+    describe('binary', () => {
+      it('generates single binary character when no additional argument was provided', () => {
+        const binary = faker.number.binary();
+        expect(binary).toBeTypeOf('string');
+        expect(binary).toHaveLength(1);
+        expect(/^[01]+$/.test(binary)).toBe(true);
+      });
+
+      it('generates a random binary string', () => {
```
Comment (ST-DDT) [inline test/number.spec.ts]: Same here

File: test/number.spec.ts (hunk 1 above)
Comment (pladreyt) [reply test/number.spec.ts]: Sure!

File: test/number.spec.ts (hunk 2 above)
Comment (pladreyt) [reply test/number.spec.ts]: True. Will update

**Comment by pladreyt** [comment]
> Could you please move the methods above the hex method so that the methods are a bit more in order. Binary(2), Octal(8), Hex(16)
> 
> Should we also create these methods in the stri g module to generate them in a specific length?

Sure, will reorder the methods. Should I create another PR to add them in the string module as well ?

**Comment by ST-DDT** [comment]
> Should I create another PR to add them in the string module as well ?

Yes, I think that would be the easiest solution.
The PR that gets merged later will have to add `@see` links to both though. I leave the decision to you whether you want to create the other PR now or later.

File: test/number.spec.ts [hunk 4]
```diff
@@ -11,7 +11,9 @@ describe('number', () => {
   seededTests(faker, 'number', (t) => {
     t.describeEach(
       'int',
-      'hex'
+      'hex',
+      'binary',
+      'octal'
```
Comment (ST-DDT) [inline test/number.spec.ts]: Please adjust the methods here as well.

File: test/number.spec.ts [hunk 5]
```diff
@@ -240,6 +242,72 @@ describe('number', () => {
       });
     });
 
+    describe('binary', () => {
+      it('generates single binary character when no additional argument was provided', () => {
+        const binary = faker.number.binary();
+        expect(binary).toBeTypeOf('string');
+        expect(binary).toHaveLength(1);
+        expect(binary).toMatch(/^[01]+$/);
```
Comment (ST-DDT) [inline test/number.spec.ts]:
```suggestion
        expect(binary).toMatch(/^[01]$/);
```

File: test/number.spec.ts [hunk 6]
```diff
@@ -240,6 +242,72 @@ describe('number', () => {
...
+        }).toThrowError(`Max ${max} should be greater than min ${min}.`);
+      });
+    });
+
+    describe('octal', () => {
+      it('generates single octal character when no additional argument was provided', () => {
+        const octal = faker.number.octal();
+        expect(octal).toBeTypeOf('string');
+        expect(octal).toHaveLength(1);
+        expect(octal).toMatch(/^[0-7]+$/);
```
Comment (ST-DDT) [inline test/number.spec.ts]:
```suggestion
        expect(octal).toMatch(/^[0-7]$/);
```

**Review by Shinigami92 — DISMISSED** [review]
Please solve @ST-DDT suggestions, but already a good-to-go from my side

File: test/number.spec.ts (hunk 4 above)
Comment (pladreyt) [reply test/number.spec.ts]: Sure !

File: test/number.spec.ts (hunk 5 above)
Comment (pladreyt) [reply test/number.spec.ts]: > > Should I create another PR to add them in the string module as well ?
> 
> Yes, I think that would be the easiest solution. The PR that gets merged later will have to add `@see` links to both though. I leave the decision to you whether you want to create the other PR now or later.

Will create the other PR probably later today and add the `@see` links

**Review by ST-DDT — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**